### PR TITLE
chore: a small script to export agribalyse

### PR DESCRIPTION
**Pull request by alexgarel**
A simple script to get a CSV of agribalyse categories.

### CODE REVIEW

File: scripts/snippets/agribalyse_categories_csv.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+while agri_cats:
+    children = set(
+        tagid
+        for tagid, d in datas.items()
+        if set(d.get("parents",[])) & agri_cats
+    )
+    all_agri_cats |= children
+    agri_cats = children
+
+# get agribalyse exploring parents
```
Comment (benbenben2) [inline scripts/snippets/agribalyse_categories_csv.py:24]: alternatively, you can put comment as docstring inside the function

File: scripts/snippets/agribalyse_categories_csv.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        continue
+    agri, depth = agribalyse(tagid)
+    rows.append({
+        "tagid": tagid,
+        "cat_en": d["name"].get("en", ""),
+        "cat_fr": d["name"].get("fr", ""),
+        "agribalyse_food_code": agri,
+        })
+
+len(rows)
```
Comment (benbenben2) [inline scripts/snippets/agribalyse_categories_csv.py:55]: Is it needed?

File: scripts/snippets/agribalyse_categories_csv.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    children = set(
+        tagid
+        for tagid, d in datas.items()
+        if set(d.get("parents",[])) & agri_cats
+    )
+    all_agri_cats |= children
+    agri_cats = children
+
+# get agribalyse exploring parents
+def agribalyse(tagid, depth=0):
```
Comment (benbenben2) [inline scripts/snippets/agribalyse_categories_csv.py:25]: I would use function annotation there: https://peps.python.org/pep-0008/#function-annotations
`def agribalyse(tafid: srtr, depth: int = 0)  -> (str, int): `
and then, return a str and a int all the time

File: scripts/snippets/agribalyse_categories_csv.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            candidates.append(candidate)
+    candidates.sort(key=lambda c: c[1])
+    return candidates[0] if candidates else None
+
+
+# build data
+rows = []
+for tagid in sorted(all_agri_cats):
+    d = datas[tagid]
+    if not ({"en", "fr"} & set(d["name"].keys())):
```
Comment (benbenben2) [inline scripts/snippets/agribalyse_categories_csv.py:44]: Isn't it "in" instead of "&"?

File: scripts/snippets/agribalyse_categories_csv.py (hunk 2 above)
Comment (alexgarel) [reply scripts/snippets/agribalyse_categories_csv.py:55]: Non it's just a check :-)  I was running this in a console, really…

File: scripts/snippets/agribalyse_categories_csv.py (hunk 4 above)
Comment (alexgarel) [reply scripts/snippets/agribalyse_categories_csv.py:44]: I using the intersection of two sets.

**Comment by alexgarel** [comment]
@benbenben2  thanks a lot for reviewing. That said, it's a "snippet" (but I prefer to keep it in scripts/snippet rather than in github). So if you don't mind I wont fix "style". It's just to keep it for a time where we might need it again.

**Comment by alexgarel** [comment]
@benbenben2 that said your approval would enable merging this :-D

**Review by benbenben2 — APPROVED** [review]

**Pull request merged**